test/1.EtherSplitter.test.ts

```diff
@@ -1,20 +1,83 @@
-import {expect, use} from 'chai';
-import {Contract} from 'ethers';
-import {deployContract, MockProvider, solidity} from 'ethereum-waffle';
-import EtherSplitter from '../build/EtherSplitter.json';
+import { expect, use } from "chai";
+import { Contract } from "ethers";
+import { deployContract, MockProvider, solidity } from "ethereum-waffle";
+import EtherSplitter from "../build/EtherSplitter.json";
 
 use(solidity);
 
-describe('Ether Splitter', () => {
+describe("Ether Splitter", () => {
   const [alice, bob, charlie, david] = new MockProvider().getWallets();
   let splitter: Contract;
 
   beforeEach(async () => {
-    const contractConstructorArgs: any[] = []
-    splitter = await deployContract(alice, EtherSplitter, contractConstructorArgs);
+    const contractConstructorArgs: any[] = [bob.address, charlie.address];
+    splitter = await deployContract(
+      alice,
+      EtherSplitter,
+      contractConstructorArgs
+    );
   });
 
-  it('Deploys correctly and has an address', async () => {
-    expect(splitter.address).to.be.properAddress
+  it("Deploys correctly and has an address", async () => {
+    expect(await splitter.address).to.be.properAddress;
+  });
+
+  it("Addres is set correctly", async () => {
+    expect(await splitter.firstAddr()).to.equal(bob.address);
+    expect(await splitter.secondAddr()).to.equal(charlie.address);
+  });
+
+  it("Ether split", async () => {
+    const bobBalance = await bob.getBalance();
+    const charlieBalance = await charlie.getBalance();
+    const aliceBalance = await alice.getBalance();
+
+    const transactionValue: number = Math.floor((Math.random() * 100) / 2) * 2;
+
+    await splitter.spliter({ value: transactionValue });
+    expect(aliceBalance).to.not.equal(await alice.getBalance());
+
+    expect(await bob.getBalance()).to.equal(
+      bobBalance.add(transactionValue / 2)
+    );
+    expect(await charlie.getBalance()).to.equal(
+      bobBalance.add(transactionValue / 2)
```

Comment:
```suggestion
      charlieBalance.add(transactionValue / 2)
```

It turned out to be the same because both have the same initial balance.